### PR TITLE
feat: Use PathTableResolver for relations

### PR DESCRIPTION
**Pull request by hermanschaaf**
This change introduces the ability to generate resolvers for Go SDK-based generators. When possible, we now use `PathTableResolver` to resolve table relations.

### Previous Iteration

In a previous iteration of this PR, we generated resolver code instead of using `PathTableResolver`.

For example, instead of:

```
func fetchBasePointerRelationInnerRelations(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
     panic("not implemented")
}
```

we could now generate:

```
func fetchBasePointerRelationInnerRelations(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
	// This is a generated base implementation. You may edit it if necessary.
	p := helpers.ToPointer(parent.Item).(*tests.PointerRelationStruct)
	if p.Inner == nil {
		return nil
	}
	res <- p.Inner.Relations
	return nil
}
```

This worked in the majority of cases, but using PathTableResolver is simpler for now. We may revisit generating the functions again in the future.

### CODE REVIEW

**Comment by hermanschaaf** [comment]
We could use the `PathTableResolver` introduced in this PR: https://github.com/cloudquery/cq-provider-sdk/pull/440 instead of generating the function. This would reduce the amount of generated code, but make it harder to edit the resolvers in the cases where that is necessary.

**Comment by bbernays** [comment]
I am alright with making it a bit harder in some cases because the `PathTableResolver` also gives us protection against mistakenly accessing structs in nil pointers which is a class of bugs that we have not been able to get rid of.

Also this same argument would have precluded us from implementing the original `PathResolver`

**Comment by hermanschaaf** [comment]
@bbernays the generated code takes care of nil pointer checking, so that's solved either way. 

The difference between `PathTableResolver` and `PathResolver`, I think, is that it will more often need to be modified: basically whenever the API doesn't fill out all the fields in a given struct, or misses something important (like tags). I'm not sure how often that happens in the `PathResolver` case.

**Comment by bbernays** [comment]
@hermanschaaf - Maybe I am missing something, but it looks like it only ever checks a single level down... For example functions like this:
https://github.com/cloudquery/cq-provider-aws/blob/01efd1af29e56e5a8c6aae9b87ab63868451c426/resources/services/cloudfront/distributions.go#L840-L854

**Comment by hermanschaaf** [comment]
@bbernays Hmm, I don't understand, it must be me who's missing something 😄 The function you linked seems to be checking multiple levels down?

**Comment by bbernays** [comment]
Sorry that was an example function, it was manually generated, but looking at the cq-gen template, it looks like it is only able to do 1 level of checking

**Comment by hermanschaaf** [comment]
@bbernays oh no, the generated code does (or at least should) go multiple levels down

**Comment by bbernays** [comment]
Well I am glad it was me that was missing that rather than the code!

**Comment by hermanschaaf** [comment]
@roneli @bbernays I've updated this PR to start using `PathTableResolver`. It simplifies things quite a lot, since we no longer need to keep track of whether fields are pointers. Please take a look when you get a chance :bow:

File: codegen/builder.go [hunk 1]
```diff
@@ -127,6 +130,14 @@ func (tb TableBuilder) BuildTable(parentTable *TableDefinition, resourceCfg *con
 		return nil, err
 	}
 
+	// base field becomes this table
+
```
Comment (roneli) [inline codegen/builder.go:134]: nit: rogue new line

**Review by roneli — COMMENTED** [review]
Looks good, had 2 comments (1 minor minor nit ^^ forgive me 🙇 )

File: codegen/config/resource.go [hunk 2]
```diff
@@ -37,9 +37,6 @@ func decodeResourceBody(ctx *hcl.EvalContext, body hcl.Body, labels []string) (*
 	if attr, exists := content.Attributes["disable_pluralize"]; exists {
 		diags = append(diags, gohcl.DecodeExpression(attr.Expr, ctx, &resource.NoPluralize)...)
 	}
-	if attr, exists := content.Attributes["description_path_parts"]; exists {
-		diags = append(diags, gohcl.DecodeExpression(attr.Expr, ctx, &resource.DescriptionPathParts)...)
-	}
```
[the remark this answers is not in the document]
Comment (hermanschaaf) [reply codegen/config/resource.go]: This was a mistake from a previous iteration, good catch, thank you!

File: codegen/builder.go (hunk 1 above)
Comment (hermanschaaf) [reply codegen/builder.go]: removed, thanks 🚮

**Review by bbernays — APPROVED** [review]

**Pull request merged**